lib/active_shipping/carriers/fedex.rb

```diff
@@ -475,7 +475,8 @@ def parse_rate_response(origin, destination, packages, response, options)
 
       if success
         rate_estimates = xml.root.css('> RateReplyDetails').map do |rated_shipment|
-          service_code = rated_shipment.at('ServiceType').text
+          service_code = rated_shipment.at('ServiceType').try(:text)
+          raise ActiveShipping::ResponseError, "The response did not contain service type" unless service_code.present?
```

Comment:
As mentioned in slack I would offer up the suggestion of rather than directly raising the `ActiveShipping::ResponseError`, create a `RateResponse` with `success = false` similar to how it is performed for no rates here https://github.com/heinzf/active_shipping/blob/master/lib/active_shipping/carriers/fedex.rb#L500

This way the `ActiveShipping::ResponseError` will have the failed `RateResponse` with the xml and request. Would just want to verify that those fields are being logged somewhere.